clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -745,6 +746,53 @@ static target getAccessTarget(const ClassTemplateSpecializationDecl *AccTy) {
       AccTy->getTemplateArgs()[3].getAsIntegral().getExtValue());
 }
 
+/// Compute a unique name that is consumable by sycl-xocc
+static std::string computeUniqueSYCLXOCCName(StringRef Name,
+                                             StringRef Demangle) {
+  /// XOCC has a maximum of 64 character for the name of the kernel function
+  /// plus the name of one parameter.
+  /// Those caracters need to be used wisely to prevent name collisions.
+  /// Its is also useful to use a name that is understandable nby the user.
+  /// so we add only 8 character of hash an only if needed.
+  constexpr unsigned MaxXOCCSize = 30;
+
+  std::string Result;
+
+  for (char c : Demangle) {
+    if (c == ':' || c == ' ' || c == '<' || c == '>' || c == ',') {
+      if (!Result.empty() && Result.back() == '_')
```

Comment:
```suggestion
      // Do not repeat _ in the cleaned-up name
      if (!Result.empty() && Result.back() == '_')
```
thanks for the quizz!